### PR TITLE
allow specificying options to the http connection manager

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

**Comment by yuval-k** [comment]
i want to make sure that we all agree on this, and then i'll write test \ docs

File: projects/gloo/api/v1/plugins/hcm/hcm.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+package hcm.plugins.gloo.solo.io;
+
+option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1/plugins/hcm";
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/wrappers.proto";
+
+option (gogoproto.equal_all) = true;
+
+message HCMSettings {
```
Comment (ilackarms) [inline projects/gloo/api/v1/plugins/hcm/hcm.proto]: i would prefer a rename to `HttpConnManagerSettings` or something like that

**Comment by ilackarms** [comment]
one comment, else lgtm

**Comment by yuval-k** [comment]
can't use any till fix fix proto vendoring, as the plugin proto will not be serialized.

**Comment by yuval-k** [comment]
opened #596 instead

**Pull request closed without merging**